style_editor.html

```diff
@@ -9912,15 +9967,111 @@
+          <button 
+            id="SETTINGS_BUTTON" class="icon-button" title="Settings"
+            onclick="toggleSettingsPanel()">
+            <i class="fa-solid fa-cog"></i> Settings</button>
       </span>
+        <div id="settings_panel" class="settings-panel"
+          title="Turning on checkboxes saves your settings for future visits to the Style Editor page.">
+
+          <button
+           title="Close Settings Panel."
+            class="close-settings" onclick="toggleSettingsPanel()">X</button>
+
+            <h2 class="settings-header-label">Settings</h2>
+
+          <div class="settings-section">
+            <h4 class="settings-section-label" title="These settings apply to the blade preview above.">Blade Preview:</h4>
+
+            <label title="Simulates In-Hilt LED blades, (no addressable pixel effects).">
+              <input type="checkbox" id="INHILT_BUTTON" name="inhilt" onclick="ClickInhilt()">
+              <span>Inhilt LED</span>
+            </label>
+
+            <label title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on.">
+              <input type="checkbox" id="SLOW_BUTTON" name="slow" onclick="ClickSlow()">
+              <span>Slow</span>
+            </label>
+          </div>
+        </div>
+
+<script>
+
+function toggleSettingsPanel() {
+  var settingsButton = FIND("SETTINGS_BUTTON");
+  var settingsPanel = FIND("settings_panel");
+
+  settingsPanel.classList.toggle("show");
+
+  // Mouseleave event listener
+  if (settingsPanel.classList.contains("show")) {
+    var timeoutId = null;
+    settingsPanel.addEventListener("mouseleave", function(e) {
+      timeoutId = setTimeout(function() {
+        settingsPanel.classList.remove("show");
+      }, 1000);
+    });
+    settingsPanel.addEventListener("mouseenter", function(e) {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+        timeoutId = null;
+      }
+    });
+  }
+}
+
+/* Globally declare all the elements for Settings */
+var inhiltButton = FIND("INHILT_BUTTON"), inhiltState;
+var slowButton = FIND("SLOW_BUTTON"), slowState;
+
+/* Settings buttons saved as local storage */
+function getSavedState(buttonState, defaultValue) {
+  var value = localStorage.getItem(buttonState);
+  console.log("Retrieved SavedState for " + buttonState + ": " + value);
+
+  if (value === null) {
+    return defaultValue;
+  }
+  if (buttonState === "wavlenValue") {
+    return parseInt(value);
+  }
+  // Convert the string to boolean unless it's null or "false"
+  return value !== "false";
+}
+
+function saveState(buttonState, settingIsOn) {
+  localStorage.setItem(buttonState, settingIsOn);
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('DOMContentLoaded', onPageLoad);
+
+function onPageLoad() {
+  // INHILT_BUTTON
+  inhiltState = getSavedState("inhiltStateSave", false);
```

Comment:
I really think we need to wrap all the savestate stuff in a class, something like:

var all_saved_states = [];

class SavedState {
  constructor(name, def, update_function) {
    this.name = name;
    this.def = def;
    this.update_function;
    all_saved_states.add(this);
  }
  onload() {
     set(getSavedState(this.name+"Save", this.def);
  }
  void set(value) {
     this.value = value;
     FIND(this.name+"_button").checked = value ? true ; false;
     saveState(this.name+"Save", value);
     this.update_function(value);
  }
  get() { return this.value; }
}

To be used like:

inhiltState = new SavedState("inhilt", false, (on) => { STATE_NUM_LEDS = on ? 1 : 144; });

Reply:
Nice.
Let me wrap my head around how to apply this.